src/borg/testsuite/__init__.py

```diff
@@ -256,6 +256,30 @@ def wait_for_mount(self, path, timeout=5):
             time.sleep(.1)
         raise Exception('wait_for_mount(%s) timeout' % path)
 
+    @contextmanager
+    def read_only(self, path):
+        """Some paths need to be made read-only for testing
+
+        Using chmod to remove write permissions is not enough due to
+        the tests running with root privileges. Instead, the folder is
+        rendered immutable with chattr or chflags, respectively.
+        """
+        if sys.platform.startswith('linux'):
+            cmd_immutable = 'chattr +i "%s"' % path
+            cmd_mutable = 'chattr -i "%s"' % path
+        elif sys.platform.startswith('darwin'):
+            cmd_immutable = 'chflags schg "%s"' % path
+            cmd_mutable = 'chflags noschg "%s"' % path
```

Comment:
BSD has `chflags` to make folders immutable so tests should now be working on OS X, too.

Reply:
Yup, (even though the job eventually times out) the relevant tests pass:

>     .tox/py35/lib/python3.5/site-packages/borg/testsuite/archiver.py::ArchiverTestCase::test_readonly_list 
>     [gw0] [ 10%] PASSED .tox/py35/lib/python3.5/site-packages/borg/testsuite/archiver.py::ArchiverTestCase::test_readonly_list 
>     .tox/py35/lib/python3.5/site-packages/borg/testsuite/archiver.py::ArchiverTestCase::test_readonly_mount 
>     [gw0] [ 11%] PASSED .tox/py35/lib/python3.5/site-packages/borg/testsuite/archiver.py::ArchiverTestCase::test_readonly_mount 

https://travis-ci.org/github/borgbackup/borg/jobs/667056832